### DISCUSSION TITLE
Support showing docstrings for extension modules

### DISCUSSION THREAD

**Discussion by segevfiner** [microsoft/pylance-release#3011] [Enhancement]
Reopening https://github.com/microsoft/pylance-release/issues/2076 as the suggestion to add the docstrings to pyi files isn't really helpful as no existing pyi file does that since no one wants to manually duplicate all the doc strings.

Pylance currently does not show docstrings for extension modules even if they have type stubs. For example:

`spam.pyx`:
```cython
# cython: language_level=3str

def f() -> int:
    """This is f."""
    return 123
```

`spam.pyi`:
```python
def f() -> int: ...
```

Build with: `cythonize -i spam.pyx`, and then create some new Python file and then import and try to get completions for spam. You will see the function `f` but it won't show its docstring.

**Comment by judej** [comment]
We don't show docstrings from compiled modules. You could add a `.pyi` file with the docstrings.

Will add this to discussion as a potential enhancement and wait for upvotes.

**Comment by segevfiner** [comment]
P.S. @judej  If you want feature request to go to discussions, you can configure it to do so as part of the GitHub repository config, some VS Code repositories have done so.
